src/app/services/data.service.ts

```diff
@@ -420,64 +420,94 @@ export class DataService {
   }
 
   async getPatientEgfrInfo(patientId): Promise<boolean> {
-    const egfrChartData: ChartDataSets = { data: [], label: 'eGfr', fill: false };
-    const xAxisLabels: string[] = [];
     this.egfr = emptyEgfr;
     this.egfr.tableData = [];
     this.egfr.chartData = [];
     this.goalsdataservice.getPatientEgfr(patientId)
-      .pipe(
-        finalize(() => {
-          this.egfr.chartData.push(egfrChartData);
-          this.egfrDataSource.data = this.egfr.tableData;
-          const vsLowDateRow: EgfrTableData = (this.egfr.tableData.reduce((low, e) =>
-            reformatYYYYMMDD(low.date) < reformatYYYYMMDD(e.date) ? low : e));
-          const vsHighDateRow: EgfrTableData = (this.egfr.tableData.reduce((high, e) =>
-            reformatYYYYMMDD(high.date) >= reformatYYYYMMDD(e.date) ? high : e));
-          this.egfr.mostRecentEgfr.date = vsHighDateRow.date;
-          this.egfr.mostRecentEgfr.value = vsHighDateRow.egfr;
-          this.egfr.mostRecentEgfr.unit = vsHighDateRow.unit;
-          this.egfr.mostRecentEgfr.test = vsHighDateRow.test;
-          this.egfr.mostRecentEgfr.result = formatEgfrResult(vsHighDateRow.egfr, vsHighDateRow.unit);
-          const minDate = new Date(moment(vsLowDateRow.date.toString()).startOf('month').format('MMMM DD YYYY H:mm A'));
-          this.egfr.suggestedMin = minDate;
-          const maxDate = new Date(moment(vsHighDateRow.date.toString()).add(1, 'M').startOf('month').format('YYYY-MM-DD hh:mm:ss'));
-          this.egfr.suggestedMax = maxDate;
-          const lineChartOptions = getLineChartOptionsObject(10, 70, this.egfr.suggestedMin, this.egfr.suggestedMax);
-          const lineChartAnnotations = getEgrLineChartAnnotationsObject();
-          this.egfr.lineChartOptions = { ...lineChartOptions, annotation: lineChartAnnotations };
-          this.egfr.xAxisLabels = [];
-          let yr = '';
-          let prevYr = '';
-          this.egfr.tableData.map(vs => {
-            if (moment(vs.date.toString()).format('YYYY') !== prevYr) {
-              yr = moment(vs.date.toString()).format('YYYY');
-              prevYr = yr;
-            } else {
-              yr = '';
-            }
-            // @ts-ignore
-            xAxisLabels.push([moment(vs.date.toString()).format('MMM'),
-            moment(vs.date.toString()).format('DD'),
-              yr]
-            );
-          });
-          this.egfr.xAxisLabels = xAxisLabels;
-        })
-      )
       .subscribe(res => {
-        this.egfr.tableData.push(res);
         const egfr = {
           x: new Date(res.date),
           y: res.egfr
         };
-        // @ts-ignore
-        egfrChartData.data.push(egfr);
+        if (!this.aggregatedChartData.find(x => x.label === res.test)) {
+          this.aggregatedChartData.push({
```

Comment:
So is it possible that we might have duplicate labels, for in this case does the label include the code?

Reply:
The label should include the code unless it doesn't match the string we are looking for.  You can see how that value is being created in the "formatEGFRCode" function found here.  https://github.com/MccCareplan/providersmartapp/pull/210/files#diff-1e63ad9b0fb0e1ff3a6ff0176ddddd19c2b741f4b6596cec31aa8ad574a67d55R170

Reply:
Ok - Thought that might be the case but wanted to make sure - Since generally we use codes to ensure uniqueness. Looks good then.